edgexfoundry/device-sdk-go
# feat!: Make All ProtocolDriver interface API implementations required

BREAKING CHANGE: Move Discover(), ValidateDevice() to the ProtocolDriver interface to that they are required like the existing interfaces. This forces at a minimum an empty implementation which gives exposure to the developer that they exist.

Close #1374







**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/device-sdk-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break) all device service need to implement the Discover(), ValidateDevice() func
  <link to docs PR>

## Testing Instructions

- Run unit test
- Run core services and the simple device service to verify API still works

## New Dependency Instructions (If applicable)
